### PR TITLE
[CARBONDATA-3497] Support to write long string for streaming table

### PR DESCRIPTION
**Pull request by zzcclp**
Support to write long string for streaming table.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? No
 
 - [ ] Any backward compatibility impacted? No
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/364/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/364/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/364/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/365/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/365/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/365/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/367/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/367/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/369/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/369/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/367/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/369/

**Comment by zzcclp** [comment]
@QiangCai @jackylk @ravipesala This pr is ready, please help to review, thanks.

File: hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java [hunk 1]
```diff
@@ -387,7 +392,12 @@ protected void readRowFromStream() {
         }
       } else {
         if (isNoDictColumn[colCount]) {
-          int v = input.readShort();
+          int v = 0;
+          if (dimensionDataTypes[colCount] == DataTypes.VARCHAR) {
```
Comment (QiangCai) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java]: better to use a boolean array to store the result.

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithLongString.scala [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+import org.apache.spark.sql.test.util.QueryTest
+import org.scalatest.BeforeAndAfterAll
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.metadata.schema.table.CarbonTable
+import org.apache.carbondata.core.util.CarbonProperties
+import org.apache.carbondata.core.util.path.CarbonTablePath
+import org.apache.carbondata.streaming.parser.CarbonStreamParser
+
+case class StreamLongStrData(id: Integer, name: String, city: String, salary: java.lang.Float,
```
Comment (QiangCai) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithLongString.scala]: how long does this test case take?

Comment (zzcclp) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithLongString.scala]: It's about 1min 10sec, test reports show that TestStreamingTableWithLongString took 1min 10sec to run.

File: hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java (hunk 1 above)
Comment (zzcclp) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/382/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/382/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/382/

**Comment by zzcclp** [comment]
@QiangCai Please review, thanks.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/390/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/390/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/389/

File: hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java [hunk 3]
```diff
@@ -163,6 +164,14 @@ public StreamRecordReader(QueryModel mdl, boolean useRawRow) {
             .getDirectDictionaryGenerator(storageColumns[i].getDataType());
       }
     }
+    dimensionsIsVarcharTypeMap = new boolean[dimensionCount];
+    for (int i = 0; i < dimensionCount; i++) {
+      if (storageColumns[i].getDataType() == DataTypes.VARCHAR) {
```
Comment (QiangCai) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java]: dimensionsIsVarcharTypeMap[i]=storageColumns[i].getDataType() == DataTypes.VARCHAR

Comment (zzcclp) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/419/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/421/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/421/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**